packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -114,6 +121,8 @@ export class ServiceBuilderImpl implements ServiceBuilder {
     app.use(notFoundHandler());
     app.use(errorHandler());
 
+    app.use(this.handlers);
```

Comment:
The ordering here is a bit tricky. Looking from outside this class I would've expected the middleware to be added just before the routes. Tbh I'd be up for replacing the `.addRouter` method with a single overloaded `.use` method that behaves just like the express one with 1 or 2 args. Then you'd at least end up with the flexibility of adding middleware before and after the plugin routes.

Reply:
Agreed. This feels more like an opportunity for generalization rather than feature addition

Reply:
yeap,  you are right,  and this does not even work,  because the notFoundHandler() should be called last